packages/x-charts/src/internals/plugins/corePlugins/useChartInteractionListener/useChartInteractionListener.ts

```diff
@@ -0,0 +1,154 @@
+'use client';
+import * as React from 'react';
+import {
+  createUseGesture,
+  dragAction,
+  pinchAction,
+  wheelAction,
+  moveAction,
+  hoverAction,
+} from '@use-gesture/react';
+import { ChartPlugin } from '../../models';
+import {
+  UseChartInteractionListenerSignature,
+  AddInteractionListener,
+  AddMultipleInteractionListeners,
+  ChartInteraction,
+  ChartInteractionHandler,
+} from './useChartInteractionListener.types';
+
+type ListenerRef = Map<ChartInteraction, Set<ChartInteractionHandler<any, any>>>;
+
+const preventDefault = (event: Event) => event.preventDefault();
+
+// Create our own useGesture hook with the actions we need
+// Better for tree shaking
+const useGesture = createUseGesture([
+  dragAction,
+  pinchAction,
+  wheelAction,
+  moveAction,
+  hoverAction,
+]);
+
+export const useChartInteractionListener: ChartPlugin<UseChartInteractionListenerSignature> = ({
+  svgRef,
+}) => {
+  const listenersRef = React.useRef<ListenerRef>(new Map());
+
+  const retriggerEvent = React.useCallback((interaction: ChartInteraction, state: any) => {
```

Comment:
Should we be more specific about the type of `state`? 

I don't think it will work if `state` is a string, for example. It needs to be an object, right?

Reply:
There is not a lot of need tbh, this is only used in the hook below to re-trigger the events.

Reply:
By the way `re-trigger` is a bit confusing. Should it be `transmitingTrigger`. Because for me `re-trigger` means it got trigger once, and I trigger it a second time.